specification/azsadmin/resource-manager/update/Microsoft.Update.Admin/preview/2021-07-01/Updates.json

```diff
@@ -0,0 +1,461 @@
+          },
+          {
+            "$ref": "Update.json#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "UpdateLocations.json#/parameters/UpdateLocationParameter"
+          },
+          {
+            "$ref": "Updates.json#/parameters/UpdateParameter"
+          },
+          {
+            "$ref": "Update.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "UpdateRuns.json#/definitions/UpdateRun"
+            }
+          },
+          "202": {
+            "description": "ACCEPTED",
+            "schema": {
+              "$ref": "UpdateRuns.json#/definitions/UpdateRun"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true
+      }
+    }
+  },
+  "definitions": {
+    "Update": {
+      "description": "Information pertaining to an update.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "description": "Properties of an update.",
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/UpdateModel"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "Update.json#/definitions/Resource"
+        }
+      ]
+    },
+    "UpdateModel": {
+      "description": "Properties of an update.",
+      "type": "object",
+      "properties": {
+        "displayName": {
+          "description": "Name of the update.",
+          "type": "string"
+        },
+        "description": {
+          "description": "Description of the update.",
+          "type": "string"
+        },
+        "version": {
+          "description": "Version of the update.",
+          "type": "string"
+        },
+        "state": {
+          "description": "Current state of the update.",
+          "$ref": "#/definitions/UpdateState"
+        },
+        "availabilityType": {
+          "description": "The availability type of the update. The location where the update is available.",
+          "$ref": "#/definitions/AvailabilityType"
+        },
+        "releaseLink": {
+          "description": "Release link about the update.",
+          "type": "string"
+        },
+        "packagePath": {
+          "description": "Path to the update package.",
+          "type": "string"
+        },
+        "packageSizeInMb": {
+          "description": "Size of update package.",
+          "type": "integer",
+          "format": "int64"
+        },
+        "publisher": {
+          "description": "Publisher of the update.",
+          "type": "string"
+        },
+        "prerequisites":{
+          "description": "List of prerequisites.",
+          "type": "array",
+		  "items": {
+            "$ref": "#/definitions/Prerequisite"
+          }
+        },
+        "updateStateProperties":{
+          "description": "Properties when update is in a specific state.",
```

Comment:
I have comments later that I figure this state should be made more generic.  The issue isn't that we need it to give the group a new name to hold the "downloadPercent", but that we need a generic updateStateProperties that could be represented in the UX properly to summarize progress (regardless of state).

Reply:
I like the idea of making these two fields more generic. downloadPercentage only represents the downloading stage, 
"progressPercentage" sounds more generic and extensible, it can be reused in Preparing, installing, healthchecking as well, even that's sounds over engineering, but if there is a requirement, it it easy to implement. 
the "installedDate" is just for backward compatibility, to make it more generic I would like to rename it as 'completionDate'. The stable states after a running state can have this value. For example when health check operation is done, we can append the completionDate.